crates/iceberg/src/expr/predicate.rs

```diff
@@ -406,13 +635,239 @@ mod tests {
 
     #[test]
     fn test_predicate_negate_set() {
-        let expression = Reference::new("a").is_in(HashSet::from([Datum::long(5), Datum::long(6)]));
+        let expression = Reference::new("a").is_in([Datum::long(5), Datum::long(6)]);
 
-        let expected =
-            Reference::new("a").is_not_in(HashSet::from([Datum::long(5), Datum::long(6)]));
+        let expected = Reference::new("a").is_not_in([Datum::long(5), Datum::long(6)]);
```

Comment:
This test case is unclear to me. Perhaps we should directly compare it with the actual result? It's possible that both `is_in` and `is_not_in` were implemented incorrectly.

Reply:
Sorry, which one? Do you mean the `negate` case or `bind` case?

Reply:
Ok, I got it. Predicate itself needs `Reference` inside. It's not something like `Result`.